encrypted-media-respec.html

```diff
@@ -605,8 +605,56 @@
+              whether consent is required for a specific configuration and whether such consent may also apply to other configurations. For example,
+              consent to one configuration may also imply consent for less powerful, more restricted configurations. Equally, a denial of consent for
+              one configuration may imply denial of consent for more powerful, less restricted configurations.
+              </p>
+              <p>
+              Supported configurations, including supported audio and video codecs, may depend on availability of optional capabilities such as
+              <a def-id="distinctive-identifier"></a>s and persistent state. The following algorithm iteratively tries to find a configuration
+              that is both supported and has user consent, each iteration considering a stronger set of restrictions derived from user consent prompts.
+              </p>
+              <p>
+              User Agents should persist consent responses at least for the duration of the <a def-id="requestMediaKeySystemAccess"></a>
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>supported configuration</var> be <code>ConsentDenied</code>.</p>
+              </li>
+              <li>
+                <p>Let <var>restrictions</var> be the empty set.</p>
+              </li>
+              <li>
+                <p>Repeat the following step while <var>supported configuration</var> is <code>ConsentDenied</code>:</p>
+                <ol>
+                  <li>
+                    <p>
+                    Let <var>supported configuration</var> and, if provided, <var>restrictions</var> be the result of executing the
+                    <a def-id="get-supported-configuration-and-consent-algorithm"></a> algorithm
+                    with <var>implementation</var>, <var>candidate configuration</var>, <var>restrictions</var> and <var>origin</var>.
+                    </p>
+                  </li>
+                </ol>
+                <li>
+                  <p>Return <var>supported configuration</var>.</p>
+                </li>
+              </li>
+            </ol>
+          </section>
+          <section>
+            <h5>Get Supported Configuration and Consent</h5>
+            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>,
+              <var>restrictions</var> and <var>origin</var>, this algorithm returns a supported configuration, <code>NotSupported</code>, or <code>ConsentDenied</code> as appropriate and, in the <code>ConsentDenied</code> case, <var>restrictions</var>.
+            </p>
+            <p class="note"><var>restrictions</var> consists of an implementation-specific set of configurations for which consent has been denied by the user.</p>
```

Comment:
See other references to "configurations" and this line in my comments below.

I think it's possible to deny a configuration without denying everything in that configuration in future iterations. For example, if an audio codec happens to be in an _accumulated configuration_ that is denied because "optional" was converted to "required", that doesn't mean that the audio codec should be disallowed in future iterations.